lib/msal-node/src/cache/CacheManager.ts

```diff
@@ -0,0 +1,94 @@
+import { Storage } from './Storage';
+import { Serializer, Deserializer, InMemoryCache, JsonCache } from '@azure/msal-common';
+import { ICachePlugin } from './ICachePlugin';
+
+const defaultSerializedCache: JsonCache = {
+    Account: {},
+    IdToken: {},
+    AccessToken: {},
+    RefreshToken: {},
+    AppMetadata: {}
+};
+
+
+export class CacheManager {
+
+    hasChanged: boolean;
+    storage: Storage;
+    persistence: ICachePlugin;
+
+    constructor(storage: Storage, cachePlugin?: ICachePlugin) {
+        this.hasChanged = false;
+        this.storage = storage;
+        this.storage.registerChangeEmitter(this.handleChangeEvent.bind(this))
+        if (cachePlugin) {
+            this.setPersistence(cachePlugin);
+        }
+
+    }
+
+    setPersistence(persistence: ICachePlugin): void {
+        this.persistence = persistence;
+    }
+
+    handleChangeEvent() {
+        this.hasChanged = true;
+    }
+
+    mergeState(oldState: JsonCache, currentState: JsonCache) {
+        // TODO
+        // mergeUpdates(old, new)
+        // mergeRemovals(old, new)
+        return {
+            ...oldState,
+            ...currentState
+        };
+    }
+
+    //TODO think about separating serialize / deserialize from writeToPersistance and readFromPersistance
+
+    async serialize(): Promise<string> {
+        const cache: string = JSON.stringify(Serializer.serializeAllCache(this.storage.getCache()));
+        if (this.persistence) {
+            await this.persistence.writeToStorage((stateFromDisk) => {
+                const jsonFromDisk = JSON.parse(stateFromDisk);
+                return JSON.stringify(
+                    this.mergeState(
+                        jsonFromDisk,
```

Comment:
is jsonFromDisk of the format `JsonCache`? I think so but would like to test this once. Also if the disk cache is empty JSON.parse() would not return `{}`, it would return `null`; we may need to handle it.

Reply:
it is json cache but that type is not exposed to dev 
, I think extension portion should handle what to pass here since it knows the state of the file. 
I changed to initialize to empty object and throw on bad parse